### ISSUE TITLE
Bug: Frontend and Backend doesn't handle required in JsonSchema the same way

### ISSUE THREAD

**Issue by inkhey** [tracim/tracim#4121]
## Description and expectations

Frontend and Backend doesn't check required the same way:
- Frontend does split schema according to the different section to show (explained in ui schema by "tracim:display_group") and split only required at the upper level now. So required for sub-item is not guarantee to work.
- Backend currently check all the jsonschema, so check all required in one pass.

This situation lead to issue when dealing with required as backend for example will refuse to accept partial update of the jsonschema that doesn't add all the required.

To solve the issue related to backend doesn't do partial check on required, we can:
1. Make backend do a split like frontend : + complex, + validation
2. Make backend do a validation on schema without required field :  - complex, -validation
3. Make default value mandatory for required filed:  -- complex, +validation, - remove some specific use case of the required. (this solution permit also to remove the split in frontend and to handle properly the nested required)

### Version information

- Tracim version: 3.5 pre-release

**Comment by raphj** [comment]
I don't think we should purposefully ignore the required field when validating.
If we don't support required correctly, we should tell people not to use it I think.

**Comment by lebouquetin** [comment]
@inkhey partial update is not exclusive with required fields.

**Comment by lebouquetin** [comment]
@inkhey as discussed with @Skylsmoi several times, supporting required fields is not in the target for Tracim 3.5. We wont fix this for this release.

**Comment by raphj** [comment]
For now we are not supporting `required`. This needs to be added somewhere in the documentation related to the schema forms.

Should we support it at some point, any state should be valid with respect to the schema, which means that in particular, the initial state should be valid and have values for required fields.

I see two ways of achieving this:
 - force the user to fill the required fields when first filling the forms before submitting the form
 - have default values for these fields.

In any case this requires some careful thought about the UX of this issue.
